fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineShortcuts.kt

```diff
@@ -0,0 +1,316 @@
+            )
+        }
+
+        item {
+            AddEngineButton(onAddEngineClicked = onAddEngineClicked)
+        }
+    }
+}
+
+@Composable
+private fun Title(title: String) {
+    Box(
+        modifier = Modifier
+            .height(48.dp)
+            .fillMaxWidth(),
+        contentAlignment = Alignment.CenterStart,
+    ) {
+        Text(
+            text = title,
+            color = FirefoxTheme.colors.textAccent,
+            fontWeight = FontWeight.W400,
+            modifier = Modifier.padding(horizontal = 16.dp),
+            style = FirefoxTheme.typography.headline8,
+        )
+    }
+}
+
+@Suppress("LongParameterList", "LongMethod")
+@Composable
+private fun SearchItem(
+    engine: SearchEngine,
+    name: String,
+    isEnabled: Boolean,
+    onCheckboxClicked: (SearchEngine, Boolean) -> Unit,
+    onEditEngineClicked: (SearchEngine) -> Unit,
+    onDeleteEngineClicked: (SearchEngine) -> Unit,
+) {
+    val isMenuExpanded: MutableState<Boolean> = remember { mutableStateOf(false) }
+
+    Row(
+        modifier = Modifier
+            .defaultMinSize(minHeight = 56.dp)
+            .padding(start = 4.dp),
+    ) {
+        Checkbox(
+            modifier = Modifier.align(Alignment.CenterVertically),
+            checked = isEnabled,
+            onCheckedChange = { onCheckboxClicked.invoke(engine, it) },
+            colors = CheckboxDefaults.colors(
+                checkedColor = FirefoxTheme.colors.formSelected,
+                uncheckedColor = FirefoxTheme.colors.formDefault,
+            ),
+        )
+
+        Spacer(modifier = Modifier.width(20.dp))
+
+        Image(
+            modifier = Modifier
+                .align(Alignment.CenterVertically)
+                .size(24.dp),
+            bitmap = engine.icon.asImageBitmap(),
+            contentDescription = stringResource(
+                id = R.string.search_engine_icon_content_description,
+                engine.name,
+            ),
+        )
+
+        Spacer(modifier = Modifier.width(16.dp))
+
+        Text(
+            modifier = Modifier
+                .align(Alignment.CenterVertically)
+                .padding(vertical = 8.dp)
+                .weight(1f),
+            text = name,
+            style = FirefoxTheme.typography.subtitle1,
+            color = FirefoxTheme.colors.textPrimary,
+        )
+
+        if (engine.type == SearchEngine.Type.CUSTOM) {
+            Box(
+                modifier = Modifier
+                    .align(Alignment.CenterVertically),
+            ) {
+                IconButton(
+                    onClick = {
+                        isMenuExpanded.value = true
+                    },
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_menu),
+                        contentDescription = stringResource(id = R.string.content_description_menu),
+                        tint = FirefoxTheme.colors.iconPrimary,
+                    )
```

Comment:
optional nit: I think this conditional would be easier to parse if we extracted the true case `engine.type == SearchEngine.Type.CUSTOM` into a separate Composable. So the `Box` and everything in it, basically

Reply:
I think I want to revisit that comment during the compose refactoring [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1837113#c2). Made a note about your comment.